app/scripts/controllers/console.js

```diff
@@ -232,5 +281,107 @@ angular.module('console', ['sql'])
       }
     };
   })
+  .directive('lazyLoadScroll', function ($window) {
+    return {
+      restrict: 'A',
+      link: function (scope, element, attrs) {
+        var e = element[0];
+        $($window).bind('scroll', function () {
+          if (this.pageYOffset + this.innerHeight > e.scrollHeight) {
+              scope.$apply(attrs.lazyLoadScroll);
+          }
+        });
+      }
+    };
+  })
+  .directive('formattedArray', function ($filter) {
+    return {
+      restrict: 'E',
+      scope: {
+        array: '=',
+        typesarray: '=',
+        expand: '='
+      },
+      templateUrl: 'views/formatted-array.html',
+      link: function($scope) {
```

Comment:
it's a bit confusing to have `$scope` injected in the link function, the standard way is to use `scope` https://www.bennadel.com/blog/2716-when-to-use-scope-vs-scope-in-angularjs.htm